tests/unittests_test.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2017 SignalFx, Inc. All rights reserved.
+
+from httmock import all_requests, urlmatch, HTTMock
+# from urllib.parse import urlparse
+import json
+import unittest
+import signalfx.signalflow.ws
+import struct
+
+responses_map = {
+    'GET_DETECTOR': {
+        'path': '/v2/detector/abc123',
+        'content': open('tests/fixtures/get_detector_abc123.json').read(),
+        'status_code': 200,
+        'method': 'GET',
+    },
+    'GET_DETECTOR_INCIDENTS': {
+        'path': '/v2/detector/abc123/incidents',
+        'content': '[]',
+        'status_code': 200,
+        'method': 'GET',
+    },
+    'GET_INCIDENTS': {
+        'path': '/v2/incident',
+        'content': open('tests/fixtures/get_incidents.json').read(),
+        'status_code': 200,
+        'method': 'GET',
+        'params': 'include_resolved=True&offset=0',
+    },
+    'GET_INCIDENT': {
+        'path': '/v2/incident/abc123',
+        'content': open('tests/fixtures/get_incident_abc123.json').read(),
+        'status_code': 200,
+        'method': 'GET',
+    },
+    'CLEAR_INCIDENT': {
+        'path': '/v2/incident/abc123/clear',
+        'content': '',
+        'status_code': 200,
+        'method': 'PUT',
+    }
+}
+
+# Generates a function that can be used with HTTMock to test the endpoint.
+# It will validate the params, URL and Method, whilst returning a spec response.
+def mock_maker(name):
+    @all_requests
+    def mock_responder(url, request):
+        spec = responses_map.get(name)
+        if spec is None:
+            raise Exception("Unknown mock")
+
+        if spec['path'] != url.path or request.method != spec['method']:
+            return {
+                'content': 'Unknown URL' + url.path,
+                'status_code': 400,
+                }
+
+        if spec['params'] is not None and spec['params'] != url.query:
```

Comment:
`spec.get('params')`?  Only `GET_INCIDENTS` has them set.

Reply:
Changed, to `get`, but not sure I follow the ask. Yes, only that one case has them set hence the `None` check.

Reply:
Right, just referring to the resulting key error with the direct access.